### DISCUSSION TITLE
KeeShare loses directory structure

### DISCUSSION THREAD

**Discussion by markusHaferkamp** [keepassxreboot/keepassxc#10256] [Q&A] [answered]
I am running a three-computer setup with different OS's. All three machines are running KeePass 2.7.6, and have their own local databases. They all have access to a shared OwnCloud folder with a KeeShare master sync database. On every machine, the /sync/ folder is set to synchronise via KeeShare.

### Directory structure:
```
Root
|_ Sync
|    |_ Mail Clients
|    |_ Webservices
|    |_ ...
|_ Recycle bin
```

### Issues:

1. Sub-directory structure is lost on update; all synchronised data is moved to the /sync/ folder.
2. Newly-created or edited sub-directories are not pushed to the KeeShare DB.

### Comments:

- OwnCloud reports a lot of conflicts and creates `-conflicted` files if the database is open on more than one machine.
- Closing the DBs on all but one machine, creating subfolders, having that machine sync to KeeShare and _only then_ opening the DB on other machines still lead to a complete loss of subfolder structure.
- The [KeepassXC manual](https://keepassxc.org/docs/KeePassXC_UserGuide#_database_sharing_with_keeshare) mentions group structure is not shared. I'd be fine manually moving newly created entries, however, it is actively _removing_ group structure at every update. 

### Settings:
Automatically reload database on external modification
Automatically save database after every change

_(Posting here instead of on the issue tracker because I'm 99.9 % sure this one's user error and not on KeePass' side.)_

**Comment by droidmonkey** [comment, marked as answer]
#3045

**Reply by markusHaferkamp** [reply]
Ah. "Group" was the magical keyword I'd been missing, thank you!

I remain mildly confused after reading that thread — shouldn't local group structure stay intact? I (now) know group structure is not synchronised (reading the manual would've helped, mea culpa). But still the "whole sync group is flattened every time something is updated" seems like it's unexpected?
